src/DependencyInjection/Compiler/ExtensionCompilerPass.php

```diff
@@ -257,14 +261,30 @@ private function isSubtypeOf(string $type, string $subject, string $class): bool
             case 'global':
                 return true;
             case 'instanceof':
+                if (!class_exists($subject)) {
+                    return false;
+                }
+
                 $subjectReflection = new \ReflectionClass($subject);
 
                 return $classReflection->isSubclassOf($subject) || $subjectReflection->getName() === $classReflection->getName();
             case 'implements':
+                if (!interface_exists($subject)) {
+                    return false;
+                }
+
                 return $classReflection->implementsInterface($subject);
```

Comment:
in those cases (except the first one) you could return both expressions : if + current return